reana_workflow_controller/consumer.py

```diff
@@ -240,46 +232,6 @@ def _update_job_progress(workflow_uuid, msg):
                     job.status = job_status
 
 
-def _update_job_cache(msg):
```

Comment:
Hmm, whilst it is true that caching has been disabled since a long time, we may want to resurrect and retest it one day perhaps... Hence let me muse a bit on this. 

Does this code actually runs on the server-side? We have cache=off by default on the client side, so I think this code should not even be triggered on the server-side. If it is not, then we don't seem to have a problem here? If it is, then we should fix it so that it would not be? (or inactivate it via a server-side configuration settings without removal of the code).

That said, philosophically speaking, I like [suckless approach](https://suckless.org/philosophy/) arguing that _"the more code lines you have removed, the more progress you have made"_ :wink: Removing old inactive code is a good thing, especially if we investigate something like Armada or Kueue soon... Then it might be a perfect time for cleaning job cache everywhere. Until then, perhaps "inactivation" may still be having some advantages.

Reply:
Honestly, there is a lack of documentation for cache internals so I have no idea how it works or where it is configured exactly. Need to dig into the code. 

Doesn't look like we have a cache config in the `helm` chart for this. Also, user docs only have a few things mentioned like the "CACHE=off" parameter for r-client. There are some Python tests for cache.

Looks like only the `serial` engine supports cache. Other engines do not check if a job is cached from the `job-controller`.

I heard that cache was not working so, in my opinion, we should remove it. Otherwise, this is just a dead code. But, I agree, maybe, I should create a separate issue to remove the cache and we can address it later (espcecially if we plan work on improving consumers and internals)